packages/split-layout/test/split-layout.common.js

```diff
@@ -73,6 +73,96 @@ describe('split layout', () => {
+            <div id="first">primary</div>
+          </vaadin-split-layout>
+        `);
+        await nextRender();
+        expect(layout.querySelector('#first').getAttribute('slot')).to.be.equal('primary');
+        expect(layout.querySelector('#second').getAttribute('slot')).to.be.equal('secondary');
+      });
+
+      it('should assign a slot if only element has "primary" slot pre-defined', async () => {
+        const layout = fixtureSync(`
+          <vaadin-split-layout>
+            <div id="second">secondary</div>
+            <div id="first" slot="primary">primary</div>
+          </vaadin-split-layout>
+        `);
+        await nextRender();
+        expect(layout.querySelector('#first').getAttribute('slot')).to.be.equal('primary');
+        expect(layout.querySelector('#second').getAttribute('slot')).to.be.equal('secondary');
+      });
+
+      it('should respect assigned slot if only one element has slot pre-defined after order is inverted', async () => {
+        const layout = fixtureSync(`
+          <vaadin-split-layout>
+            <div id="second">secondary</div>
+            <div id="first" slot="primary">primary</div>
+          </vaadin-split-layout>
+        `);
+        await nextRender();
+
+        const first = layout.querySelector('#first');
+        layout.prepend(first);
+        await nextRender();
+
+        expect(layout.querySelector('#first').getAttribute('slot')).to.be.equal('primary');
+        expect(layout.querySelector('#second').getAttribute('slot')).to.be.equal('secondary');
+      });
+
+      it('should swap slots if children without pre-defined slots invert order', async () => {
+        const layout = fixtureSync(`
+          <vaadin-split-layout>
+            <div id="second">secondary</div>
+            <div id="first">primary</div>
+          </vaadin-split-layout>
+        `);
+        await nextRender();
+
+        const second = layout.querySelector('#second');
+        layout.prepend(second);
+        await nextRender();
+
+        expect(layout.querySelector('#first').getAttribute('slot')).to.be.equal('secondary');
+        expect(layout.querySelector('#second').getAttribute('slot')).to.be.equal('primary');
+      });
+
+      it('should assign slots only for direct children', async () => {
+        const layout = fixtureSync(`
+          <vaadin-split-layout>
+            <div id="first">primary</div>
+            <vaadin-split-layout id="second">
+              <div id="nested-first slot="primary"></div>
+              <div id="nested-second" slot="secondary"></div>
+            </vaadin-split-layout>
+          </vaadin-split-layout>
+        `);
+        await nextRender();
+
+        const first = layout.querySelector('#first');
+        expect(first.getAttribute('slot')).to.be.equal('primary');
+        const second = layout.querySelector('#second');
+        expect(second.getAttribute('slot')).to.be.equal('secondary');
```

Comment:
Should this test actually test that `nested-first` doesn't get a slot? Or I might've not understood what this test is intended to cover.

Reply:
The purpose of this test is to check that the nested split-layout gets the right slot. This was added after this [comment](https://github.com/vaadin/web-components/pull/6938#discussion_r1420268607) was made because I was using `this.querySelector('[slot]')` and that case would cause an issue. Now, I am using the `children` property, so that issue wouldn't happen.

Reply:
Got it 👍